### PR TITLE
Feature/nats metrics

### PR DESCRIPTION
**Pull request by jeevanjoseph**
- Enables metrics collection for NATS server.
- adds the Prometheus exporter as an additional container in the NATS pod. 
- the exporter exposes the `/metrics` endpoint that exposes the metrics in prometheus exposition format.

### CODE REVIEW

File: deploy/complete/helm-chart/mushop/charts/nats/values.yaml [hunk 1]
```diff
@@ -5,6 +5,11 @@ image:
   tag: 2.1.2
   pullPolicy: IfNotPresent
 
+exporter: 
+  repository: synadia/prometheus-nats-exporter
```
Comment (mvandervliet) [inline deploy/complete/helm-chart/mushop/charts/nats/values.yaml:9]: Is this the preferred way/image to export nats metrics?

Comment (jeevanjoseph) [reply deploy/complete/helm-chart/mushop/charts/nats/values.yaml:9]: Yes it is. [Docs](https://github.com/nats-io/prometheus-nats-exporter#run) use the same image as well.
Synadia is the "commercial" face of the folks who build NATS.io. 
It simply runs the `prometheus-nats-exporter`. 

We could replace this with our own `Dockerfile` or use the [bitnami image](https://hub.docker.com/r/bitnami/nats-exporter) which is Apache 2.0 licensed.

File: deploy/complete/helm-chart/mushop/charts/nats/templates/nats-deployment.yaml [hunk 2]
```diff
@@ -22,6 +22,9 @@ spec:
         run: {{ include "nats.name" . }}
       annotations:
         sidecar.istio.io/rewriteAppHTTPProbers: "true"
+        prometheus.io/scrape: "true"
```
Comment (junior) [inline deploy/complete/helm-chart/mushop/charts/nats/templates/nats-deployment.yaml:25]: scrape or exporter?

Comment (jeevanjoseph) [reply deploy/complete/helm-chart/mushop/charts/nats/templates/nats-deployment.yaml:25]: scrape - `prometheus-nats-exporter` simply provides a scrape endpoint. The helm chart version can configure prom using the prom operator, in any case, the mechanism is always scrape.

**Pull request merged**